### PR TITLE
feat(datepicker): add deprecation notices

### PR DESCRIPTION
**Pull request by wesleycho**
- Add deprecation notices for datepicker & popup

### CODE REVIEW

File: src/datepicker/datepicker.js [hunk 1]
```diff
@@ -707,14 +735,50 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
...
+        scope.datepickerOptions.datepickerPopupTemplateUrl :
+        datepickerPopupConfig.datepickerPopupTemplateUrl;
+      datepickerTemplateUrl = angular.isDefined(scope.datepickerOptions.datepickerTemplateUrl) ?
+        scope.datepickerOptions.datepickerTemplateUrl : datepickerPopupConfig.datepickerTemplateUrl;
+      altInputFormats = angular.isDefined(scope.datepickerOptions.altInputFormats) ?
+        scope.datepickerOptions.altInputFormats :
+        datepickerPopupConfig.altInputFormats;
+    } else {
+      if (datepickerPopupAttributeWarning) {
+        $log.warn('uib-datepicker-popup attributes deprecated and will be removed in UI Bootstrap 1.3, use datepicker-options attribute instead');
```
Comment (Foxandxss) [inline src/datepicker/datepicker.js]: This sentence needs a word. Perhaps `attributes are deprecated and ...`.

**Comment by Foxandxss** [comment]
this LGTM, bugs will arose for sure but we will fix them before 1.3

Also, this needs a massive doc update (to reflect how to use the directive right now with some deprecation notice).

I will create a doc PR once we merge this.

**Comment by Foxandxss** [comment]
Ah, also the demo needs to be updated. That can be done in another PR or here.

File: src/datepicker/datepicker.js (hunk 1 above)
Comment (wesleycho) [inline src/datepicker/datepicker.js]: Whoops - will fix that quick before merging.

**Comment by wesleycho** [comment]
Let's get this in - I actually wanted to do more in this PR, but this alone turned out to be a lot of work.

**Comment by Foxandxss** [comment]
Merge it. I will do the rest.

**Pull request closed without merging**